types/src/check_summed_hex.rs

```diff
@@ -86,22 +86,26 @@ pub fn encode_iter(input: &(impl AsRef<[u8]> + ?Sized)) -> impl Iterator<Item =
     })
 }
 
-fn string_is_uppercase(input: &(impl AsRef<[u8]> + ?Sized)) -> bool {
+pub fn string_is_same_case(input: &(impl AsRef<[u8]> + ?Sized)) -> bool {
     let bytes: &[u8] = input.as_ref();
-    bytes
+    let case_counts = bytes
         .iter()
         .map(|byte| *byte as char)
         .filter(|c| c.is_alphabetic())
-        .all(|c| c.is_uppercase())
-}
+        .fold((0_u64, 0_u64), |(upper_count, lower_count), c| {
+            if c.is_uppercase() {
+                (upper_count + 1, lower_count)
+            } else {
+                (upper_count, lower_count + 1)
+            }
+        });
 
-fn string_is_lowercase(input: &(impl AsRef<[u8]> + ?Sized)) -> bool {
-    let bytes: &[u8] = input.as_ref();
-    bytes
-        .iter()
-        .map(|byte| *byte as char)
-        .filter(|c| c.is_alphabetic())
-        .all(|c| c.is_lowercase())
+    match case_counts {
+        (0, 0) => true,
+        (0, 0..=u64::MAX) => true,
+        (0..=u64::MAX, 0) => true,
+        (0..=u64::MAX, 0..=u64::MAX) => false,
+    }
```

Comment:
No need to count; and we should return early as soon as we found one upper- and one lowercase letter.
Or use `Itertools`:

```rust
bytes.iter()
    .filter_map(|byte| {
        let c = *byte as char;
        c.is_alphabetic().then(|| c.is_uppercase())
    })
    .tuple_windows()
    .all(|(upper0, upper1)| upper0 == upper1)
```